module6/lessons/Week3/BinaryTrees.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: page
+title: Binary Trees - CS Topic - Data Structure
+---
+
+## Introduction
+
+Today's lesson on the binary tree data structure comes to you in three parts. 
+
+**Part 1**: Independent research
+
+**Part 2**: Class walkthrough of key operations and time complexity
+
+**Part 3**: Diving into a C# implementation of a binary tree
```

Comment:
I would change this to remove the reference to C# because you're letting them find any language.

Reply:
Actually I might remove this section entirely because I think it might be confusing because this lesson has more than 3 parts. Maybe change to something more vague such as. "Today's lesson on binary trees will be a mixture of independent research, class walkthrough and discussion, and diving into a code implementation."